### ISSUE TITLE
Is there any way to catch failed DNS lookups?

### ISSUE THREAD

**Issue by vgrinko** [ariya/phantomjs#13245]
Hello, I am using Phantomjs to catch 404/500/etc errors (specifically I am using response.status value within "onResourceReceived" callback). Although it does not allow to catch "DNS Lookup Error", and I can't find other way to do so. Can anyone please help?

Thank you!

**Comment by Vitallium** [comment]
PhantomJS can report DNS errors. But, I think you've encountered with a wrong behavior of resource callbacks that I found some time ago.
You can read more here: https://groups.google.com/forum/#!topic/phantomjs-dev/jpVIVSyRv2g

Also, could you provide a minimal test case?

**Comment by vgrinko** [comment]
Thank you @Vitallium, you are absolutely right, I was looking at wrong status values, so I was able to adjust my code accordingly.
Although, I noticed, that in some cases status is "null". For example, when I am trying to open non-existing domain. Here is *response* object within *onResourceReceived* callback:
```
{"contentType":null,"headers":[],"id":1,"redirectURL":null,"stage":"end","status":null,"statusText":null,"time":"2015-05-20T01:34:21.698Z","url":"http://blahblahblah123456.com/"}
```
And I don't see "NULL" value neither in QT docs nor in the QT sources:
http://doc.qt.io/qt-4.8/qnetworkreply.html
https://github.com/ariya/phantomjs/blob/master/src/qt/qtbase/src/network/access/qnetworkreply.h#L67

Do you know if there is any documentation which explains "NULL" status?

Thank you for your help!

**Comment by vgrinko** [comment]
I have some update on this issue.
The tool which I am building is a sort of crawler. In addition to capturing network errors, it can abort some of them based on the filter function passed to the config. Don't ask why :)
So, after some investigation on what is causing "null" requests in "onResourceRequested" I discovered the reason:

when I abort the request in "onResourceRequested" callback, it is then being reported in "onResourceReceived" as 
```
{"contentType":null,"headers":[],"id":1,"redirectURL":null,"stage":"end","status":null,"statusText":null,"time":"2015-05-20T01:34:21.698Z","url":"http://blahblahblah123456.com/"}
```
At the same time "onResourceError" callback reports it as: 
```
{"errorCode":301,"errorString":"Protocol \"\" is unknown","id":1,"url":""}
```

So what I ended up doing is recording *ids* of all aborted requests and ignoring these aborted requests in onResourceError callback. Also, I switched to using "onResourceError" for all my error tracking needs instead of "onResourceRequested".

It solves my problem, but  it is really unclear why onResourceError is reporting *301 "Protocol \"\" is unknown" error.*
